docs/articles/movie_recommendation_using_vectordb.markdown

```diff
@@ -0,0 +1,568 @@
+import lancedb
+import pandas as pd
+from lancedb.pydantic import LanceModel, Vector
+
+data = pd.read_csv("movie_embeddings.csv")
+data.drop(columns=["movie_index"], inplace=True)
+
+movie_data = []
+for index, row in data.iterrows():
+    embedding_vector = ast.literal_eval(row["genre_embeddings"])
+    movie_data.append(
+        {
+            "title": row['title'],
+            "embeddings": embedding_vector,
+            "genre_labels": row['genre_labels']
+        }
+    )
+
+# Define LanceDB model
+class Movie(LanceModel):
+    title: str
+    embeddings: Vector(128)  
+    genre_labels: str
+
+# Create LanceDB connection
+db = lancedb.connect("./db")
+movie_table = db.create_table(
+    "movies", 
+    schema=Movie,
+    mode="Overwrite")
+movie_table.add(movie_data)
+```
+
+The code might seem overwhelming at first glance, but it's not as complex as it appears. Essentially, we start by reading the essential data from our CSV file. Then, we establish a connection to LanceDB to set up our table and add our movie data
+
+Each row in the table corresponds to a single movie, with columns storing data such as title, genres, overview, and embeddings. This columnar format offers a convenient way to store and retrieve information for a variety of tasks related to embeddings. 
+
+## Using Genre Spectrum Embeddings to get the relevant recommendations.
+
+And now, after all the groundwork, we've arrived at the final piece of the puzzle. Let's generate some relevant recommendations using `genre_embeddings` and LanceDB VectorDB.
+
+```python
+movie_data_pd = pd.DataFrame(movie_data)
+def get_recommendation(title):
+    result = (
+        movie_table.search(movie_data_pd[movie_data_pd["title"] == title]["embeddings"].values[0]).metric('cosine')
+        .limit(5)
+        .to_pandas()
+    )
+    return result
+
+result = get_recommendation("Toy Story")
+result[['title']]
+```
+
+Well let me break down some things for you.
+ 
+First, we initialize our LanceDB table containing our relevant data. Then, for a given movie title, we check if we have this title in our dataset. If it exists, we perform a cosine similarity search on all other movies, returning the top 5 most relevant movie titles.
+
+And just like that, you've crafted an impressive movie recommendation system that can swiftly provide you with the best selection of movies. This is made possible by leveraging `genre_embeddings` created with the assistance of a neural network and storing them in LanceDB vector database.
+
+Doing all this will provide you with a list of recommended movies.
+
+*Just to clarify, I've employed a rudimentary deep learning model architecture to showcase how we can improve our recommendation system. While the movie recommendations may not yield optimal results, the team at Tubi would be the experts to provide you with the exact model architecture they've utilized for developing this system.*
+
+That being said, Here's an overview of the entire training code..
```

Comment:
I don't think the entire code is needed here if you link to a colab with the code, and parts are there in the text.

Reply:
Done!